pytorch_pfn_extras/training/extension.py

```diff
@@ -222,3 +224,60 @@ def decorator(ext: 'ExtensionLike') -> 'ExtensionLike':
 
 def _as_extension(ext: 'ExtensionLike') -> Extension:
     return ext if isinstance(ext, Extension) else _WrappedExtension(ext)
+
+
+class ExtensionEntry:
+    """Extension and options.
+
+    Args:
+        extension: An extension.
+        name: Name of extension. If omitted, copied from the attributes of
+            ``extension`` argument.
+        priority: Invocation priority of the extension.
+            Default is ``PRIORITY_READER``.
+        trigger: Trigger object that determines when to invoke the extension.
+            Default is ``(1, 'iteration')``.
+        call_before_training: Flag to call extension before training.
+            Default is ``False``.
+
+    .. seealso::
+       :meth:`pytorch_pfn_extras.training.ExtensionsManager.extend`
+    """
+
+    def __init__(
+            self,
+            extension: 'ExtensionLike',
+            *,
+            name: Optional[str] = None,
+            priority: Optional[int] = None,
+            trigger: Optional['TriggerLike'] = None,
```

Comment:
I think it's nice to use `Optional['TriggerLike']` here, so could you remove `Optional` from TriggerLike definition?
https://github.com/pfnet/pytorch-pfn-extras/blob/a8d02a665de0dfaa6f142a065d7ba6ab5c13b41d/pytorch_pfn_extras/training/_trigger_util.py

Reply:
`ExtensionEntry.__init__` interprets `trigger=None` as omitted option, whereas `get_trigger` interprets it as `never_trigger`.

Reply:
Ah, I see, thanks for clarification 😄